src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -172,7 +183,26 @@
            {:state :starting}))
 
        (init-error [this]
-         @init-error)))))
+         @init-error)
+
+       PoolLifetime
+       (block-until-ready [this]
+         (log/info (trs "Blocking execution until db init pool has finished"))
+         (deref pool-future))
```

Comment:
pffft I forgot how futures work -- won't get evaluated twice, which is what I was concerned about. This is fine.